### PR TITLE
[libc][docs] adds threads.h impl status docs

### LINKED ISSUE #88067: [libc][docs] generate docs for threads.h
Using [n3096](https://www.open-std.org/jtc1/sc22/wg14/www/docs/n3096.pdf) Section 7.28, we should fill out the function list in a new file called libc/utils/docgen/threads.json (so that we have entries about which standard section talks about what, similar to libc/utils/docgen/fenv.json).

Then we should run ./libc/utils/docgen/docgen.py threads.h > libc/docs/threads.rst to create a page for threads.h.

Then we should add a link to threads in libc/docs/index.rst so that it's linked from libc.llvm.org (similar to fenv).

We need to add [the include for |check|](https://github.com/llvm/llvm-project/commit/e127997155a1cd1c5692c42aad074064c8dad099) to be recognized.

https://github.com/llvm/llvm-project/issues/87835 and https://github.com/llvm/llvm-project/pull/87946 are recent examples

### PR DESCRIPTION
**Pull request by ashfordium**
Closes #88067.

Adding @nickdesaulniers for review.

### CODE REVIEW

File: libc/utils/docgen/threads.json [hunk 1]
```diff
@@ -0,0 +1,89 @@
+{
+  "macros": {
+    "ONCE_FLAG_INIT": {
+      "defined": "7.28.1.3",
+      "implemented": true
```
Comment (nickdesaulniers) [inline libc/utils/docgen/threads.json:5]: mind dropping the implemented field for now?

**Review by nickdesaulniers — COMMENTED** [review]
Rest LGTM; thanks for filing this out!

File: libc/utils/docgen/threads.json (hunk 1 above)
Comment (Flandini) [reply libc/utils/docgen/threads.json]: For sure, removed.

**Review by nickdesaulniers — APPROVED** [review]
need us to merge this for you?

**Pull request merged**